### PR TITLE
UI updates

### PR DESCRIPTION
**Pull request by bdlukaa**
Created some UI updates. Now the app feels better :)

UI updates and fixes:

- 🦜 Language section in settings can now be collapsed and Language Tiles now have the flag of the country
- 🖇 The name of some songs was being clipped. (TODO: the name of albums and artists are also being clipped)
- 💫 A weird scroll behavior. (TODO: apply the scroll behavior of the fixed listview to the whole app using a `ScrolConfiguration`)
- 👨‍🎤 Ink well wasn't working on the albums section
- 🤳 Ink well didn't include the image

| Before | Now |
| ------- | ----- |
| previous_language | now_language | 
| name_clipping | name_not_clipping |
| weird_scroll_behavior | cool_scroll_behavior |
| ink_well_not_working | Not fixed. TODO: fix it |
| songs_ink_well | ink_well_working |

I also added an animated icon to the current track FAB pause-play button. When tap the button, it animates. (TODO: the button is not animated when the song is paused from another source, such as notifications)

### CODE REVIEW

**Comment by bdlukaa** [comment]
Updated the things that were left to do:

- **FIXED** the name of albums and artists are also being clipped
- **FIXED** the button is not animated when the song is paused from another source, such as notifications

Minor things added:

- Added a `>` in the about section (settings)
- Added a 🔽 in the language section (settings)
- Animated the icon button (🔼) in the `now playing` screen to follow the container animation

I also added more todos and reviews. You can search in your workspace the todos (enable matchcase):

- TODO(bdlukaa)
- TODO(alex)
- review(alex)

I also formatted the files I edited using [dartfmt](https://dart.dev/tools/dartfmt)

File: lib/screens/settings/miscellaneous.dart [hunk 1]
```diff
@@ -1,41 +1,36 @@
...
 import 'package:harmonoid/language/constants.dart';
 
 import 'package:harmonoid/screens/settings/settings.dart';
 import 'package:harmonoid/scripts/configuration.dart';
 import 'package:harmonoid/scripts/states.dart';
 
-
 class MiscellaneousSetting extends StatefulWidget {
-  MiscellaneousSetting({Key key}) : super(key: key);
+  const MiscellaneousSetting({Key key}) : super(key: key);
```
Comment (alexmercerind) [inline lib/screens/settings/miscellaneous.dart:9]: Didn't know constructors can be `const` in Dart.
Cool!

File: lib/screens/settings/language.dart [hunk 2]
```diff
@@ -14,77 +13,305 @@ class LanguageSetting extends StatefulWidget {
...
-          RadioListTile(
-            value: LanguageRegion.deDe,
-            title: Text(LanguageRegion.deDe.data[0]),
-            subtitle: Text(LanguageRegion.deDe.data[1]),
-            groupValue: this.languageRegion,
-            onChanged: (Object object) => this.setLanguageRegion(object),
+        ),
+        child: AspectRatio(
+          aspectRatio: 16 / 9,
+          child: Flag(languageRegion.countryCode, fit: BoxFit.cover),
```
Comment (alexmercerind) [inline lib/screens/settings/language.dart:114]: This is really cool addition! Really making app vibrant.

**Review by alexmercerind — APPROVED** [review]
Awesome PR @bdlukaa!

Takes away nearly all the hard edges.

Initially I made myself many trade-offs by trimming the name of songs & albums so that they fit properly. Good that you removed.

As you could possibly tell by the code... I can't stay without things like defining parameter type definition in the method definition. `this` is something that I use everywhere even in C++... I'm so used to it. 😅

Earlier code of app used JSONs all around the app (so hard  to manage), now I introduced single classes like `Track`, `Album`, `Playlist` etc. to easily manage and see their properties without having to remember anything.

Thanks a lot.
Appreciate your work! (a lot)

File: lib/screens/settings/miscellaneous.dart (hunk 1 above)
Comment (bdlukaa) [reply lib/screens/settings/miscellaneous.dart:9]: They're used so the widget don't get rebuilt when a parent widget is rebuilt. Used to increase performance. I recommend to add `const` everywhere possible in your app

**Pull request merged**